### ISSUE TITLE
[Feature]: Allow users to add username & password with OAuth 2.0 authentication type

### ISSUE THREAD

**Issue by pranavkanade** [appsmithorg/appsmith#11959]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

A user wants to use a username & password along with Client ID & Client Secret in their authenticated API data source.
As of today we only allow Client ID & Client Secret if the Authentication type is set to OAuth 2.0

### Why should this be worked on?

User request -
> We are using OAuth2 for authentication with the client ID, client secret, user name, and password. It works very well with Postman and Postman has the required fields. I've looked at several lo-code and no-code platforms and they don't seem to accommodate these credentials with OAuth2. It would be great if it worked with Appsmith.

**Comment by hiteshjoshi** [comment]
@pranavkanade the idea of Auth 2.0 is to not send username and password. May I know what they are using as provider for auth. Also if any details on the use case?

**Comment by Nikhil-Nandagopal** [comment]
Yeah this needs more details. Even postman doesn't support this. @pranavkanade can you please ask the user where they are seeing this on postman?
![Uploading Screenshot 2022-03-18 at 4.04.45 PM.png…]()

**Comment by ame-appsmith** [comment]
This is the information sent by the user:

> Here's the Postman screenshot. Each customer has a "Client ID" and "Client Secret" and we can provide one or more credentials ("User Name" and "Password") per customer. These are useful because we can configure authorization rules at the OAuth2 layer, e.g. one user for integration can access/do some stuff, while another user only has read-only access.
> 
> The credentials get sent to the OAuth2 layer and return the access token, which is added to the "Authentication" attribute in the header in all subsequent requests. We also add an enterprise ID to the header to map the requests to the right customer / tenant.
> 
> Notice the "Grant Type" is "Password Credentials" that enables these additional fields in the "Configure New Token" form.

![image](https://user-images.githubusercontent.com/101268423/159093421-58c5118e-cf9f-46e6-8e12-40abbb6b9062.png)

**Comment by Nikhil-Nandagopal** [comment]
@ame-appsmith thank you for the information!
